pallets/asset/src/lib.rs

```diff
@@ -1150,11 +1150,15 @@ impl<T: Config> Module<T> {
     fn ensure_agent_with_custody_and_perms(
         origin: T::Origin,
         ticker: Ticker,
+        portfolio_kind: PortfolioKind,
     ) -> Result<IdentityId, DispatchError> {
         let data = <ExternalAgents<T>>::ensure_agent_asset_perms(origin, ticker)?;
 
-        // Ensure the caller has not assigned custody of their default portfolio and that they are permissioned.
-        let portfolio = PortfolioId::default_portfolio(data.primary_did);
+        // Ensure the caller has not assigned custody of their portfolio and that they are permissioned.
+        let portfolio = PortfolioId {
+            did: data.primary_did,
+            kind: portfolio_kind,
+        };
         let skey = data.secondary_key.as_ref();
         Portfolio::<T>::ensure_portfolio_custody_and_permission(portfolio, data.primary_did, skey)?;
         Ok(data.primary_did)
```

Comment:
I think it would be better to return the `portfolio` here, instead of just the DID.  This would simplify the functions that call this function, since most of them have to recreate the same `portfolio` value.